app/lib/DBInit.scala

```diff
@@ -0,0 +1,121 @@
+package lib
+
+import scala.util.Try
+import scala.concurrent.Future
+import scala.concurrent.ExecutionContext.Implicits.global
+
+import cats.implicits._
+import cats.data.EitherT
+import play.api.libs.json._
+import reactivemongo.bson._
+import play.modules.reactivemongo.ReactiveMongoApi
+import reactivemongo.api.commands.{DefaultWriteResult, LastError, UpdateWriteResult}
+import reactivemongo.api.collections.bson.BSONCollection
+import com.github.nscala_time.time.Imports._
+
+import lib.Data._
+
+final case class InitialData(rooms: List[BSONDocument], screenings: List[BSONDocument])
+class DBInit(api: ReactiveMongoApi) {
+
+  def initDB(data: JsValue): EitherT[Future, Error, Unit] = for {
+    initData <- EitherT.fromOption[Future](processData(data), InvalidBody: Error)
+    _ <- insertData(initData)
+  } yield ()
+
+  def insertData(data: InitialData): EitherT[Future, Error, Unit] = EitherT(
+    for {
+      db <- api.database
+      resRooms <- data.rooms.traverse(
+        db.collection[BSONCollection]("rooms").insert(_))
+
+      resScreenings <- data.screenings.traverse(
+        db.collection[BSONCollection]("screenings").insert(_))
+
+
+    } yield (resRooms, resScreenings) match {
+      case (_: LastError, _: LastError) => Left(Unknown)
+      case _                            => Right(())
+    }
+  )
+
+  // Process data ---------------------------------------------------------------------------------
+  def processData(data: JsValue): Option[InitialData] = data match {
+    case JsObject(obj) => for {
+      jsonDate        <- obj.get("date")
+      jsonRooms       <- obj.get("rooms")
+      jsonScreenings  <- obj.get("screenings")
+
+      dateStr <- jsonDate.asOpt[String]
+      date <- Try(DateTimeFormat.forPattern("dd-MM-yyyy").parseDateTime(dateStr)).toOption
```

Comment:
This way you loose information about what is wrong (e.g. wrong date format), and as a result you return generic error `"invalid request body"`